addons/website/models/website_visitor.py

```diff
@@ -306,11 +306,27 @@ def _link_to_visitor(self, target):
 
         return target
 
-    def _cron_archive_visitors(self):
+    def _cron_clean_inactive_visitors(self):
+        """ Unlink inactive visitors (see '_inactive_visitors_domain' for details).
+
+        Visitors were previously archived but we came to the conclusion that archived visitors
+        have very little value and bloat the database for no reason. """
+
+        inactive_visitors = self.env['website.visitor'].sudo() \
+            .with_context(active_test=False) \
+            .search(self._inactive_visitors_domain())
+        inactive_visitors.unlink()
```

Comment:
Remark: now that we unlink old partners, in ``_link_to_visitor`` and that push tokens are moved on a separate table, shouldn't we unlink instead of archive the "old visitor that is a child of its true visitor" ?

Reply:
I think it's the same question as your first comment?

Let me know if it's something else.